core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -369,6 +386,21 @@ private boolean fetchSequencerPairs(int partsFound) {
         // No need to do anything, all discovered segments / wals will be deleted
     }
 
+    private long getSafeToPurgeTxn() {
+        long safeToPurgeTxn = txReader.getSeqTxn();
+        childViewSink.clear();
+        engine.getMatViewGraph().getDependentMatViews(tableToken, childViewSink);
+        for (int v = 0, n = childViewSink.size(); v < n; v++) {
+            final TableToken viewToken = childViewSink.get(v);
+            final MatViewRefreshState state = engine.getMatViewGraph().getViewRefreshState(viewToken);
+            if (state != null && !state.isPendingInvalidation() && !state.isInvalid() && !state.isDropped()) {
+                final long appliedToViewTxn = Math.max(0, state.getLastRefreshBaseTxn());
```

Comment:
nit: `state.getLastRefreshBaseTxn()` is -1 then the next line will handle it ok without this line.

Reply:
If `state.getLastRefreshBaseTxn() == -1` we'll end up with `Math.min(safeToPurgeTxn, -1)` which is `-1`. Do you mean that `getSafeToPurgeTxn()` can return `-1` and the caller code will handle it fine or it's something else?

Reply:
Seq Txn starts from 1 anyway

Reply:
Resolving this one as it's a nit and the suggestion is unclear. @eugenels if you know what's requested here, let me know.

Reply:
We have a discrepancy between the `TransactionLogCursor `implementations:
`TableTransactionLogV1Cursor `assumes `txnLo > 0`:

```
this.txnOffset = HEADER_SIZE + (txnLo - 1) * RECORD_SIZE;
```

`TableTransactionLogV2Cursor `assumes `txnLo > -1`:

```
assert txnLo > -1;
this.txn = txnLo;
```

In both cases, `getSafeToPurgeTxn `should not return -1.
What should the guard condition be **max(0, ..)** or  **max(1, ..)** ?